### PR TITLE
Invisible elements should never considered as "in the viewport"

### PR DESCRIPTION
**Pull request by tzi**
Hi @vvo!

We currently have a problem with layload/in-viewport: it fires image lazyloading on hidden images.
So we propose a fix to not fire isInViewport on hidden elements.

This come from the `getBoundingClientRect()` method. It always returns  `0, 0, 0, 0` on hidden elements (`display: none` or a parent with `display: none`).
So all hidden elements are considered as "is in the viewport" when the page is loaded.

This could be a breaking change, because there is no simple method to detect an hidden element.
We propose to handle an element with no height and no width as an hidden element.

What do you think about it?

Cheers,
Thomas.

### CODE REVIEW

**Comment by tzi** [comment]
PR updated with a fix.

Cheers,
Thomas.

File: test/fixtures/helper.js [hunk 1]
```diff
@@ -3,9 +3,8 @@ module.exports = {
   insertTest: insertTest,
   clean: clean,
   scroller: scroller,
-  wait: wait,
-  clean: clean
```
Comment (vvo) [inline test/fixtures/helper.js]: clean was not used?

File: in-viewport.js [hunk 2]
```diff
@@ -162,8 +168,8 @@ function createInViewport(container) {
       watching.push(elt);
     }
 
-    return function() {
-      watches.push(function() {
+    return function () {
```
Comment (vvo) [inline in-viewport.js]: stylistic change

**Comment by vvo** [comment]
Hello @tzi thanks for the awesome PRS. Only things I am against are:
- stylistic changes not related to the bugfix, even if legit, makes it hard to see what really changed
- no need to add the builded library in your commit, you can remove it
- no need to put a gitignore. I have a global gitignore so my modules does not requires them

Thanks a lot because you added an example and a test! This library is a bit old so tell me what you thought of the develop process?

File: test/fixtures/helper.js (hunk 1 above)
Comment (tzi) [inline test/fixtures/helper.js]: There was two `clean` key.

File: in-viewport.js (hunk 2 above)
Comment (tzi) [inline in-viewport.js]: It tried uniformize the code style more than change it, but you're right you write anonymous fonction without space. I will Fix it ;)

**Comment by tzi** [comment]
Hi @vvo!

1/ I used to left a clean things when I'm coding.
Don't know if it's a good or bad habit, but I understand it's confusing.

2/ The builded library is already commited in the master branch.
Why should I not updated it?

3/ The `.gitignore` file could be helpful for the (new) contributors, event if you have a global one.

What do you want me to do about it?
Do I revert the stylistic changes,  the build and the `.gitignore`?

This library is really useful! Big thanks for sharing it.
The code is pretty easy to read and do the job :)
But for example, it will need a bit more of re-organization to fix the issue like #7.

Cheers,
Thomas.

File: test/callback-argument-DOMElement.js [hunk 3]
```diff
@@ -4,7 +4,6 @@ describe('in-viewport watch callback', function() {
   beforeEach(h.clean);
   afterEach(h.clean);
 
-  var scrolled = false;
```
Comment (tzi) [inline test/callback-argument-DOMElement.js]: Unused variable

**Comment by vvo** [comment]
> The builded library is already commited in the master branch.
> Why should I not updated it?

I only generate it when releasing

> What do you want me to do about it?
> Do I revert the stylistic changes, the build and the .gitignore?

Only the build and gitignore need to be removed, otherwise looks very good! Will then test it myself and see if everything goes well with the lazyloader also.

Then Ill merge and publish etc.

**Comment by tzi** [comment]
Great plan!

My part is done ;)

Cheers!
Thomas.

**Comment by vvo** [comment]
Sorry it took me so long @tzi now merged with some test modifications, available as 3.1.0 and lazyload upgraded to to 3.1.0

**Comment by tzi** [comment]
Great!
Thank you a lot ;)

Thomas.

**Pull request merged**